### PR TITLE
Add CSS games subcategory under Frontend category.

### LINKED ISSUE #2258: Add CSS Games subcategory under Frontend category
### Category Name

Frontend

### Subcategory Name

CSS Games

### Learn CSS Grid, Flex and more by **Playing Games**

- [css grid garden](https://cssgridgarden.com/)
- [flex box froggy](https://flexboxfroggy.com/)
- [flukeout](https://flukeout.github.io/)
- [coding fantasy](https://codingfantasy.com/)

### Checklist

- [X] I have checked the existing [issues](https://github.com/rupali-codes/LinksHub/issues?q=is%3Aissue+)
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I have checked that this category does not exist
- [X] I have checked that this subcategory does not exist
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by JuanPabloDiaz**
## Fixes Issue

Closes #2258: Add CSS Games subcategory under Frontend

## Changes proposed

- Create a new subcategory called **CSS Games** under the frontend _category_.
- Modify the `CategoryDescriptions.ts` file to include CSS Games.
- Create a new file called `css-games.json` in the frontend database folder to include all the URLs related to CSS Games.
- Modify the `sitemap.xml` file to include CSS Games.
- Modify the `index.ts` file in the database folder to include CSS Games.
- Modify the `data.ts` file in the database folder to include CSS Games.

### Learn CSS Grid, Flex and more by **Playing Games**

- [css grid garden](https://cssgridgarden.com/)
- [flex box froggy](https://flexboxfroggy.com/)
- [flukeout](https://flukeout.github.io/)
- [coding fantasy](https://codingfantasy.com/)

## Screenshots

cssGames

### CODE REVIEW

File: components/TopBar/CategoryDescriptions.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ const categoryDescriptions: CategoryDescriptions = {
     'Colors mean the aspects of the appearance of objects and light sources that may be described in terms of hue, lightness, and saturation',
   'design-inspiration':
     'The right kind of inspiration will always help you come up with creative and unique ideas for your projects but sometimes you need some help getting inspired. Luckily, there are a TON of great resources for design inspiration.',
+  'css-games':
+    'CSS games are a fun way to learn and practice CSS. They are a great way to learn new CSS properties and techniques, and they can help you improve your skills in a fun and interactive way.',
```
Comment (CBID2) [inline components/TopBar/CategoryDescriptions.ts:16]:
```suggestion
    'Games are a fun way to learn and practice your FrontEnd development skills. They can help you learn new properties, techniques, and expand your coding knowledge.',
```

File: database/data.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export const sidebarData: ISidebar[] = [
       { name: 'images', url: '/images', resources: DB.images },
       { name: 'fonts', url: '/fonts', resources: DB.fonts },
       { name: 'colors', url: '/colors', resources: DB.colors },
+      { name: 'css games', url: '/css-games', resources: DB.cssGames },
```
Comment (CBID2) [inline database/data.ts:17]:
```suggestion
      { name: 'games', url: '/games', resources: DB.games },
```

File: database/frontend/css-games.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "url": "https://codingfantasy.com/",
+    "category": "frontend",
+    "subcategory": "css-games"
+  },
+  {
+    "name": "Css Grid Garden",
+    "description": "An interactive game designed to help you learn CSS grid layout. In this delightful game, you’ll cultivate a virtual carrot garden using CSS code.",
+    "url": "https://cssgridgarden.com/",
+    "category": "frontend",
+    "subcategory": "css-games"
```
Comment (CBID2) [inline database/frontend/css-games.json:14]:
```suggestion
    "subcategory": "games"
```

File: database/frontend/css-games.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  {
+    "name": "Flexbox Froggy",
+    "description": "An engaging game designed to help you master CSS flexbox. In this delightful challenge, your mission is to guide Froggy and friends to their lilypads by writing CSS code.",
+    "url": "https://flexboxfroggy.com/",
+    "category": "frontend",
+    "subcategory": "css-games"
```
Comment (CBID2) [inline database/frontend/css-games.json:21]:
```suggestion
    "subcategory": "games"
```

File: database/frontend/css-games.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  {
+    "name": "Flukeout",
+    "description": "An interactive game intertwining CSS learning with culinary creativity. As you advance, face CSS selector challenges. Your mission? Articulate precise CSS code for delectable outcomes.",
+    "url": "https://flukeout.github.io/",
+    "category": "frontend",
+    "subcategory": "css-games"
```
Comment (CBID2) [inline database/frontend/css-games.json:28]:
```suggestion
    "subcategory": "games"
```

File: database/index.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ export { default as accessibility } from './frontend/accessibility.json'
 export { default as angular } from './frontend/angular.json'
 export { default as animations } from './frontend/animations.json'
 export { default as colors } from './frontend/colors.json'
+export { default as cssGames } from './frontend/css-games.json'
```
Comment (CBID2) [inline database/index.ts:6]:
```suggestion
export { default as games } from './frontend/games.json'
```

File: public/sitemap.xml [hunk 7]
```diff
@@ -24,6 +24,11 @@
         <lastmod>2023-06-28T21:40:52+00:00</lastmod>
         <priority>0.8</priority>
     </url>
+    <url>
+        <loc>https://www.linkshub.dev/css-games</loc>
```
Comment (CBID2) [inline public/sitemap.xml:28]:
```suggestion
        <loc>https://www.linkshub.dev/games</loc>
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @JuanPabloDiaz. Your Pull Request is off to a great start. I'm thinking it'd be best to call the subcategory games instead of separating them by language. It creates more cohesiveness.

**Review by rupali-codes — COMMENTED** [review]
@CBID2 why have you renamed to `games`? won't this be confusing for other people?

**Comment by Anmol-Baranwal** [comment]
> @CBID2 why have you renamed to `games`? won't this be confusing for other people?

@rupali-codes Please deploy.

I second this, it can be `games` too but considering it is in frontend -> `css games` would be better because most of the game platforms are related to CSS.

**Comment by CBID2** [comment]
> > @CBID2 why have you renamed to `games`? won't this be confusing for other people?
> 
> 
> 
> @rupali-codes Please deploy.
> 
> 
> 
> I second this, it can be `games` too but considering it is in frontend -> `css games` would be better because most of the game platforms are related to CSS.

Oh I see.

**Comment by CBID2** [comment]
@rupali-codes & @Anmol-Baranwal, I approved the PR.

**Comment by JuanPabloDiaz** [comment]
NP, thank you!

**Review by aftabrehan — APPROVED** [review]
Excellent PR, @JuanPabloDiaz 👍

**Review by rupali-codes — APPROVED** [review]
lgtm

**Pull request merged**